test/python/unittest/API/APITest.py

```diff
@@ -1629,6 +1630,47 @@ def testAvailableChunks(self):
         for ext in io.file_extensions:
             self.makeAvailableChunksRoundTrip(ext)
 
+    def writeFromTemporaryStore(self, E_x):
+        E_x.store_chunk(np.array([[4, 5, 6]], dtype=np.dtype("int")),
+                        [1, 0])
+        data = np.array([[1, 2, 3]], dtype=np.dtype("int"))
+        E_x.store_chunk(data)
+
+    def writeFromTemporary(self, ext):
+        name = "../samples/write_from_temporary_python." + ext
+        write = io.Series(
+            name,
+            io.Access_Type.create
+        )
+
+        DS = io.Dataset
+        E_x = write.iterations[0].meshes["E"]["x"]
+        E_x.reset_dataset(DS(np.dtype("int"), [2, 3]))
+        self.writeFromTemporaryStore(E_x)
+        gc.collect()  # trigger removal of temporary data to check its copied
+
+        del write
+
+        read = io.Series(
+            name,
+            io.Access_Type.read_only
+        )
+
+        r_E_x = read.iterations[0].meshes["E"]["x"]
+        # self.assertEqual(len(r_E_x.available_chunks()), 2) # oddly, "1"
```

Comment:
@franzpoeschel this is interestingly raising `1` instead of `2`.

Reply:
This is because the results from `.available_chunks()` are backend-dependent. The JSON backend only sees this dataset:
```json
            "data": [
              [
                1,
                2,
                3
              ],
              [
                4,
                5,
                6
              ]
            ],
```
… and reports one chunk going from [0,0] to [2,3].

Reply:
Makes sense! :)